KILTprotocol/sdk-js
# test: add privacy enhanced attestation to CI

## fixes KILTProtocol/ticket#516
- adds CI tests for privacy enhanced attestation
 - basic privacy enhanced attestation setup w/ and w/o revocation
 - basic privacy enhanced verification setup w/ and w/o revocation

Note: A privacy enhanced verification should not reveal any unique claimer data. Unfortunately, when sending a message to the verifier, both the claimer's `senderAddress` and the `senderBoxPublicKey` are revealed. However, this is an unrelated issue and [tracked here](https://github.com/KILTprotocol/ticket/issues/609).

## How to test:
[Check CI](https://github.com/KILTprotocol/sdk-js/pull/281/checks?check_run_id=841667988#step:6:62). If you still want to run it on your machine, do any of these two

### 1) fast + dirty
```
docker pull kiltprotocol/mashnet-node:develop
docker run -d -p 9944:9944 kiltprotocol/mashnet-node:develop ./target/release/mashnet-node --dev --ws-port 9944 --ws-external --rpc-external
yarn test:integration:run AttestationPE
docker kill $(docker ps -q)
```

### 2) slow + tidy

```
docker build -t local/sdk-js .
docker-compose run sdk yarn test:integration:run AttestationPE && docker-compose down
```

## Checklist:
